daft/runners/pyrunner.py

```diff
@@ -218,6 +318,32 @@ def run_iter_tables(
         for result in self.run_iter(builder, results_buffer_size=results_buffer_size):
             yield result.partition()
 
+    @contextlib.contextmanager
+    def actor_pool_context(
+        self, name: str, resource_request: ResourceRequest, num_actors: int, projection: ExpressionsProjection
+    ) -> Iterator[str]:
+        actor_pool_id = f"py_actor_pool-{name}"
+
+        total_resource_request = resource_request * num_actors
+        admitted = self._attempt_admit_task(total_resource_request)
+
+        if not admitted:
+            raise RuntimeError(
+                f"Not enough resources available to admit {num_actors} actors, each with resource request: {resource_request}"
+            )
```

Comment:
We should, when creating the actor pools, deduct the resources from the total resource count so that we can throw an error when scheduling tasks that cannot run along with the actor pools instead of waiting (possibly for a future actor pool to finish, which would cause a deadlock)

Reply:
One example to test this is if the actor pool requests the entire memory of the system

Reply:
In the (perhaps far) future we should consider better ways to determine when to spin up an actor pool, such as when all the tasks that still need to be done have resource requests that are lower than total resources - actor pool resources.